st2common/tests/unit/services/test_packs.py

```diff
@@ -0,0 +1,193 @@
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        # removing test entry_point and metadata file in dummy pack
+        os.remove(entry_point)
+        os.remove(metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_permission_error_to_remove_resource_entry_point_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = PermissionError("No permission to delete file from disk")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(entry_point))
+
+        # asserting delete_action_files_from_pack function raises PermissionError
+        with self.assertRaises(PermissionError):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_permission_error_to_remove_resource_metadata_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = PermissionError("No permission to delete file from disk")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(metadata_file))
+
+        # asserting delete_action_files_from_pack function raises PermissionError
+        with self.assertRaises(PermissionError):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_exception_to_remove_resource_entry_point_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = Exception("Another exception occured")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(entry_point))
+
+        # asserting delete_action_files_from_pack function raises exception
+        with self.assertRaises(Exception):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_exception_to_remove_resource_metadata_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = Exception("Another exception occured")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(metadata_file))
+
+        # asserting delete_action_files_from_pack function raises exception
+        with self.assertRaises(Exception):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
```

Comment:
How do we know this is the error for the metadata file?

Reply:
This one still needs to check either `PermissionError` or `ResourceDiskFilesRemovalError` or both. `Exception` is too broad.

Reply:
@mahesh-orch You missed this one I think

Reply:
@m4dcoder I checked in recent push. It has been updated.